2_process/src/lake_sequences_mntoha.py

```diff
@@ -0,0 +1,202 @@
+import numpy as np
+import pandas as pd
+
+def all_dates_depths(lake_obs_interpolated, depths, depth_col='interpolated_depth', pad_before_days=0, pad_after_days=0, min_days=1):
+    """
+    Create a DataFrame with one column per depth and one row for every day from
+    `pad_before_days` days before the first observation until `pad_after_days`
+    after the last observation. Missing observations are filled in with np.NaN.
+
+    :param lake_obs_interpolated: DataFrame of observations for one lake, interpolated to the depths in the list `depths`.
+    :param depths: Full list of depths at which temperatures will be provided by the trained model.
+    :param depth_col: Name of DataFrame column with depth values to match up with `depths`. (Default value = 'interpolated_depth')
+    :param pad_before_days: Number of days to pad before the first observation (Default value = 0)
+    :param pad_after_days: Number of days to pad after the last observation (Default value = 0)
+    :param min_days: Minimum number of days included in output DataFrame (Default value = 1)
+    :returns: DataFrame of observations with NaN values for missing observations.
+
+    """
+
+    pad_before_days_td = np.timedelta64(pad_before_days, 'D')
+    pad_after_days_td = np.timedelta64(pad_after_days, 'D')
+    min_days_td = np.timedelta64(min_days, 'D')
+    one_day = np.timedelta64(1, 'D')
+    # buffer start_date by pad_before_days days
+    start_date = lake_obs_interpolated.date.min() - pad_before_days_td
+    # buffer end_date by pad_after_days days
+    end_date = lake_obs_interpolated.date.max() + pad_after_days_td
+    # check if enough days are included, pad if not
```

Comment:
maybe just make clear in your comment here that you're padding to the left, i.e., adding days to the start of the sequence to ensure it is long enough, rather than adding days to the end of the sequence